src/core/configureCommandFactory.js

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { queryString, stringToBoolean } from "../utils";
+import logQueryParam from "../constants/logQueryParam";
+
+export default ({
+  componentCreators,
+  createConfig,
+  coreConfigValidators,
+  logCommand,
+  logger,
+  initializeComponents,
+  setErrorsEnabled,
+  window
+}) => options => {
+  const config = createConfig(options);
+  config.addValidators(coreConfigValidators);
+  componentCreators.forEach(createComponent => {
+    const { configValidators } = createComponent;
+    config.addValidators(configValidators);
+  });
+  config.validate();
+  setErrorsEnabled(config.errorsEnabled);
+  const parsedQueryString = queryString.parse(window.location.search);
+  logCommand({
+    enabled:
+      parsedQueryString[logQueryParam] !== undefined
+        ? stringToBoolean(parsedQueryString[logQueryParam])
+        : config.logEnabled
+  });
```

Comment:
It looks like lines 33-40 could be put in the code for the loggerToolFactory, but I see that we are doing this so that the computed configuration can be logged correctly.  If we put the log line in initializeComponentsFactory after the tools are configured, then these lines can go in the loggerToolFactory.

Reply:
I think it's preferable to enable/disable logging as soon as possible after the config is validated so that any log statements (current or future) from anything in Alloy appropriately hits the console or not without the need for the author to understand whether the logger has yet to be appropriately configured. For example, if an error were to be thrown in `initializeComponentsFactory.js` during tool configuration for whatever reason, we would want to make sure the logging were appropriately configured beforehand because it would come into play when the error gets reported to the customer: https://github.com/adobe/alloy/pull/216/files/4de6267c353ca4f9791cb8951a310e931a3bf70f#diff-6ac18022080e60b6d96019ed637bae25R21

This might already be understood, but the configuration of the logger affects both the original logger that the core modules use and the loggers that are spawned for the components. It seems like `loggerToolFactory.js` should stick to dealing with managing the spawned loggers only and not the parent logger.